fairlearn/metrics/_extra_metrics.py

```diff
@@ -6,38 +6,194 @@
+
+    Returns
+    -------
+    float
+        The true negative rate for the data
+    """
+    unique_labels = _get_labels_for_confusion_matrix(y_true, y_pred, pos_label)
+    if len(unique_labels) == 1:
+        tnr = (unique_labels[0] != pos_label)
+    else:
+        tnr, fpr, fnr, tpr = skm.confusion_matrix(
+            y_true, y_pred,
+            sample_weight=sample_weight, labels=unique_labels, normalize="true").ravel()
     return tnr
 
 
-def false_positive_rate(y_true, y_pred, sample_weight=None):
-    r"""Calculate the false positive rate (also called fall-out)."""
-    tnr, fpr, fnr, tpr = skm.confusion_matrix(
-        y_true, y_pred, sample_weight=sample_weight, labels=[0, 1], normalize="true").ravel()
+def false_positive_rate(y_true, y_pred, sample_weight=None, pos_label=None):
+    r"""Calculate the false positive rate (also called fall-out).
+
+    Parameters
+    ----------
+    y_true : array-like
+        The list of true values
+
+    y_pred : array-like
+        The list of predicted values
+
+    sample_weight : array-like, optional
+        A list of weights to apply to each sample. By default all samples are weighted
+        equally
+
+    pos_label : scalar, optional
+        The value to treat as the 'positive' label in the samples. If `None` (the default)
+        then the largest unique value of the y arrays will be used.
+
+    Returns
+    -------
+    float
+        The false positive rate for the data
+    """
+    unique_labels = _get_labels_for_confusion_matrix(y_true, y_pred, pos_label)
+    if len(unique_labels) == 1:
+        fpr = 0
+    else:
+        tnr, fpr, fnr, tpr = skm.confusion_matrix(
+            y_true, y_pred,
+            sample_weight=sample_weight, labels=unique_labels, normalize="true").ravel()
     return fpr
 
 
-def false_negative_rate(y_true, y_pred, sample_weight=None):
-    r"""Calculate the false negative rate (also called miss rate)."""
-    tnr, fpr, fnr, tpr = skm.confusion_matrix(
-        y_true, y_pred, sample_weight=sample_weight, labels=[0, 1], normalize="true").ravel()
+def false_negative_rate(y_true, y_pred, sample_weight=None, pos_label=None):
+    r"""Calculate the false negative rate (also called miss rate).
+
+    Parameters
+    ----------
+    y_true : array-like
+        The list of true values
+
+    y_pred : array-like
+        The list of predicted values
+
+    sample_weight : array-like, optional
+        A list of weights to apply to each sample. By default all samples are weighted
+        equally
+
+    pos_label : scalar, optional
+        The value to treat as the 'positive' label in the samples. If `None` (the default)
+        then the largest unique value of the y arrays will be used.
+
+    Returns
+    -------
+    float
+        The false negative rate for the data
+    """
+    unique_labels = _get_labels_for_confusion_matrix(y_true, y_pred, pos_label)
+    if len(unique_labels) == 1:
+        fnr = 0
```

Comment:
`0` -> `0.0` and similarly return `float` elsewhere

Reply:
<mutter>type systems</mutter>